client/image_pull_test.go

```diff
@@ -0,0 +1,65 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+func TestImagePullServerError(t *testing.T) {
+	client := &APIClient{
+		HTTPCli: newMockClient(errorMockResponse(http.StatusInternalServerError, "Server error")),
+	}
+	_, err := client.ImagePull(context.Background(), "image_name", "image_tag", "auth")
+	if err == nil || !strings.Contains(err.Error(), "Server error") {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+}
+
+func TestImagePullWrongError(t *testing.T) {
+	client := &APIClient{
+		HTTPCli: newMockClient(errorMockResponse(http.StatusNotFound, "Image not found")),
+	}
+	_, err := client.ImagePull(context.Background(), "image_name", "image_tag", "auth")
+	if err == nil || !strings.Contains(err.Error(), "Image not found") {
+		t.Fatalf("expected an Image Not Found Error, got %v", err)
+	}
+}
+
+func TestImagePull(t *testing.T) {
+	expectedURL := "/images/create"
+
+	httpClient := newMockClient(func(req *http.Request) (*http.Response, error) {
+		if !strings.HasPrefix(req.URL.Path, expectedURL) {
+			return nil, fmt.Errorf("Expected URL '%s', got '%s'", expectedURL, req.URL)
+		}
+		if req.Header.Get("Content-Type") == "application/json" {
+			createConfig := types.NetworkCreateConfig{}
```

Comment:
No network is related. And ImagePulll will not involve a config in the request body.